### ISSUE TITLE
Questions about ROCm profiler

### ISSUE THREAD

**Issue by FindHao** [pytorch/kineto#743]
Hi @mwootton @aaronenyeshi ,

I found some interesting results for the models running on NVIDIA A100 and AMD MI210 GPUs. For example, I tested model resnext50_32x4d in [TorchBench](https://github.com/pytorch/benchmark). resnext50_32x4d obtains about 4.89X speedup on MI210. However, when I use PyTorch Profiler to profile models on MI210, the profile trace is strange. The total execution time of resnext50_32x4d is about 32ms on A100 and 7ms on MI210. But in the profile traces, the execution time is about 117ms on A100 and 106ms on MI210. I tested PyTorch 1.13.1 with CUDA 11.7 and ROCm 5.2. And the profile traces have been attached. Do you have any ideas?

Another question is that what do the GPU kernels do before the `Profiler Step` in ROCm profiling trace? These kernels take about 45s but no python calling context is shown in the trace view.

[resnext50.zip](https://github.com/pytorch/kineto/files/11021559/resnext50.zip)

**Comment by aaronenyeshi** [comment]
I wonder if @mwootton may have any ideas on this.

**Comment by mwootton** [comment]
Hi guys.  Sorry, very late to the party.

The extra 'out of range' data in the rocm profiles is fixed in current versions of kineto.  It came from my misunderstanding of the "warmup" phase.  During warmup the profiler is set to do full recording.  After the profiler is stopped anything that happened during warmup is filtered/discarded.  That filtering code was not general and was only being applied to cupti recordings.  So rocm traces had all the hip calls and kernel executions (but not autograd ranges) from the time the profiler went into 'warmup' until it was stopped.

I looked over the profiles, then look fine.  Training is 107ms / 64ms, inference 117ms / 296ms on rocm and cuda respectively.  However these times are pretty meaningless.  If you look at the gpu activity during the trace you can see that the gpu is at best 20% busy, it is just waiting on the cpu.  So all we are timing is how quickly the cpu can run through autograd operators and hip/cuda api calls.  The profiling (between autograd and kineto) adds some overhead.  Indeed this workload may be able to run at 100% gpu usage outside of the profiler and end up being 5x faster.

There are some models that use longer running kernels and can maintain 100% gpu usage despite the slower kernel enqueueing from the cpu due to the profiler overhead.  But this isn't really the point.  Normally you would use a profile to confirm that autograd is doing what you expect or to look at kernel execution times to identify kernels to optimize.

@FindHao @aaronenyeshi
